src/ng/directive/input.js

```diff
@@ -1290,6 +1290,9 @@ function createDateParser(regexp, mapping) {
           }
         });
         return new Date(map.yyyy, map.MM - 1, map.dd, map.HH, map.mm, map.ss || 0, map.sss * 1000 || 0);
+      } else {
+        // If we fail to parse the string with the ISO regex, it might be because it's the result of Date.toString()
+        return Date.parse(iso);
```

Comment:
#6754 was specifically about `min` and `max` using date objects, not the `ng-` variants. Because you are missing out on native validation if you use the `ng` variants. So I don't think the test ensures that we support what the commit claimed to support.